device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/ModuleClient.java

```diff
@@ -185,13 +185,36 @@ public static ModuleClient createFromEnvironment() throws ModuleClientException
         return createFromEnvironment(IotHubClientProtocol.AMQPS);
     }
 
+    /**
+     * Create a module client instance from your environment variables
+     * @param clientOptions The options that allow configuration of the module client instance during initialization
+     * @return the created module client instance
+     * @throws ModuleClientException if the module client cannot be created
+     */
+    public static ModuleClient createFromEnvironment(ClientOptions clientOptions) throws ModuleClientException
```

Comment:
This is a tricky constructor in practice. Right now, AMQP is the default protocol, and the modelId changes are only applicable for MQTT. ClientOptions does encapsulate the SSLContext though, which may make this constructor worthwile. Can you double check that the SSLContext is used correctly from this constructor?

Reply:
Especially check the case when the user provides certs in the environment variables

Reply:
yes right, I'll change default to MQTT. Also checked, ssl context will be set correctly with this constructor by DeviceClientConfig

Reply:
What happens if I pass in an SSLContext in the clientOptions and have extra trusted certs in my environment variables? Does the SSLContext take precedence? If so, you'll need to add a note in the javadoc about that